posts/2024-05-01-spring-boot-3.adoc

```diff
@@ -0,0 +1,232 @@
+            <deployPackages>spring-boot-project</deployPackages>
+        </configuration>
+    </plugin>
+----
++
+2. Create your Liberty server.xml by copying from the Spring Boot template
++
+[source,sh]
+----
+./mvnw clean liberty:create -Dtemplate=springBoot3
+mkdir -p src/main/liberty/config/
+cp target/liberty/wlp/usr/servers/defaultServer/server.xml src/main/liberty/config/server.xml
+----
++
+3. Add application arguments to your Spring Boot application configuration in Liberty server.xml
++
+[source,xml]
+----
+ <springBootApplication id="spring-liberty" location="thin-spring-liberty-0.0.1-SNAPSHOT.war" name="spring-liberty">
+    <applicationArgument>--server.servlet.context-path=/testpath</applicationArgument>
+    <applicationArgument>--myarg=myval</applicationArgument>
+ </springBootApplication>
+----
++
+4. Add println to echo args in output, `vi ./src/main/java/demo/SpringLibertyApplication.java`
++
+[source,java]
+----
+    public static void main(String[] args) {
+        System.out.println("Arguments array = " + java.util.Arrays.toString(args));
+        SpringApplication.run(SpringLibertyApplication.class, args);
+    }
+----
++
+5. Run
++
+Though the Liberty Maven Plugin doesn't currently have support for using dev mode (or "loose applications") when using the optimized deployment, we can still use the "lifecycle" built into the 'run' goal.  We do need to do a 'package' first to get the Spring Boot "repackage" packaging.
++
+[source,sh]
+----
+./mvnw package liberty:run
+----
++
+6.  Observe the output:
++
+[source,sh]
+----
+[INFO] Arguments array = [--server.servlet.context-path=/testpath, --myarg=myval]
+[INFO]   .   ____          _            __ _ _
+[INFO]  /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
+[INFO] ( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
+[INFO]  \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
+[INFO]   '  |____| .__|_| |_|_| |_\__, | / / / /
+[INFO]  =========|_|==============|___/=/_/_/_/
+[INFO]  :: Spring Boot ::                (v3.2.4)
+----
++
+and also
++
+[source,sh]
+----
+[INFO] [AUDIT   ] CWWKT0016I: Web application available (default_host): http://localhost:9080/testpath/
+----
++
+7. Test the application by visiting the following URL in a browser: http://localhost:9080/testpath/
++
+As before, you should see:
++
+  Hello from Spring Boot in Open Liberty!
++
+8. When you finish testing the application, stop the server by typing `<Ctrl>+C`.
++
+9. Recap
++
+The server is configured with the https://openliberty.io/docs/latest/reference/feature/springBoot-3.0.html[springBoot-3.0] feature. The liberty-maven-plugin has been added to the pom.xml along with special configuration to use the optimized Spring Boot deployment.  The app is packaged as an "executable WAR" by running the `spring-boot:repackage` goal in the `package` phase, and the app is bootstrapped via its `main()` method in `SpringLibertyApplication` passing in application arguments defined in `server.xml`.
```

Comment:
```suggestion
The server is configured with the https://openliberty.io/docs/latest/reference/feature/springBoot-3.0.html[springBoot-3.0] feature. The `liberty-maven-plugin` is added to the `pom.xml`, along with special configuration to use the optimized Spring Boot deployment.  The app is packaged as an executable WAR by running the `spring-boot:repackage` goal in the `package` phase, and the app is bootstrapped by its `main()` method in `SpringLibertyApplication` passing in application arguments defined in `server.xml`.
```

Reply:
here too, the results focus on config (how) instead of value (why)

Reply:
I get what you're saying, but I think if we back up you'll see why I didn't go into more detail.    Backing up, all our Liberty-related Spring Boot doc focuses on what I'm calling the optimized deployment.   In fact, we don't even call it an "optimized deployment"... we just kind of take for granted this is how you deploy Spring Boot to the point that we don't even mention the "standard WAR" deployment.  

So I think there's enough out there overall about the thinning, etc. advantages of the optimized deployment.  I'd be open to throwing in one more link on the subject.. but I don't think this one blog post needs to make the case in all that much detail.. it's been our story for awhile.